charts/newrelic-logging/templates/persistentvolume.yaml

```diff
@@ -0,0 +1,52 @@
+{{- if not (empty .Values.fluentBit.persistence) }}
+{{- if and (eq .Values.fluentBit.persistence.mode "persistentVolume") (not .Values.fluentBit.persistence.persistentVolume.existingVolumeClaim) }}
+{{- if and (not .Values.fluentBit.persistence.persistentVolume.dynamicProvisioning) (not .Values.fluentBit.persistence.persistentVolume.existingVolume) }}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "newrelic-logging.labels" . | indent 4 }}
+  name: {{ template "newrelic-logging.fullname" . }}-pv
+  annotations:
+  {{- if .Values.fluentBit.persistence.persistentVolume.annotations.volume }}
+{{ toYaml .Values.fluentBit.persistence.persistentVolume.annotations.volume | indent 4 }}
+  {{- end }}
+spec: 
+  accessModes:
+    - ReadWriteMany
+  capacity:
+    storage: {{ .Values.fluentBit.persistence.persistentVolume.size }}
+  storageClassName: {{ .Values.fluentBit.persistence.persistentVolume.storageClass }}
+  persistentVolumeReclaimPolicy: Delete
+  {{- if .Values.fluentBit.persistence.persistentVolume.extra.volume }}
+{{ toYaml .Values.fluentBit.persistence.persistentVolume.extra.volume | indent 2 }}
+  {{- end }}
+---
+{{- end }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "newrelic-logging.labels" . | indent 4 }}
+  name: {{ template "newrelic-logging.fullname" . }}-pvc
+  annotations:
+  {{- if .Values.fluentBit.persistence.persistentVolume.annotations.claim }}
+{{ toYaml .Values.fluentBit.persistence.persistentVolume.annotations.claim | indent 4 }}
+  {{- end }}
+spec:
+  storageClassName: {{ .Values.fluentBit.persistence.persistentVolume.storageClass }}
+  accessModes:
+    - ReadWriteMany
+{{- if .Values.fluentBit.persistence.persistentVolume.existingVolume }}
+  volumeName: {{ .Values.fluentBit.persistence.persistentVolume.existingVolume }}
+{{- else if not .Values.fluentBit.persistence.persistentVolume.dynamicProvisioning }}
+  volumeName: {{ template "newrelic-logging.fullname" . }}-pv
+{{- end }}
```

Comment:
Should we prevent somehow the user to be able to specify both `fluentBit.persistence.persistentVolume.existingVolume` and `fluentBit.persistence.persistentVolume.dynamicProvisioning`? I feel like these are contradictory options.

Reply:
I mean`dynamicProvisioning` doesn't make sense if an `existingVolume` is provided, but if someone is providing an `existingVolume` I don't think they should need to add the `dynamicProvisioning: false` as well (as it defaults to true).

Reply:
Exactly. Somehow Helm should throw an error (or a warning at least) informing of this inconsistent configuration. This validation could happen in the `tpl` file. We could use `fail`, like in [here](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_license_secret.yaml.tpl#L7-L9).

Reply:
I was saying the opposite but i'll take it.